sql/src/main/java/org/apache/druid/sql/calcite/schema/DruidSchema.java

```diff
@@ -825,6 +837,14 @@ Map<SegmentId, AvailableSegmentMetadata> getSegmentMetadataSnapshot()
     return segmentMetadata;
   }
 
+  /**
+   * @return segmentMetadataInfo. Callers can only safely read the contents of the map.
+   */
+  public ConcurrentHashMap<String, ConcurrentSkipListMap<SegmentId, AvailableSegmentMetadata>> getSegmentMetadataInfoUnsafe()
```

Comment:
As we are placing the monitor in the same package and as this is marked to be an unsafe method, it should atleast be package protected.
Seeing as this is a sensitive method,
- should we annotate it as so?
- should we return a copy of the map rather than the actual?

Reply:
> should we return a copy of the map rather than the actual?
Yeah, I think that's the right thing to do. I will have to think about whether or not each thread should get it's own copy of the map or if it's better to just keep a single copy for reading and share that among threads.